### ISSUE TITLE
Gradient for kornia.HomographyWarper with mode='nearest'

### ISSUE THREAD

**Issue by darkAlert** [kornia/kornia#827]
Apparently, the gradient is not working correctly for kornia.HomographyWarper with mode='nearest'. Because I always get gradient=0 in backward step. But it works fine when I set mode='bilinear' 
[tensorboard screenshot for mode='nearest'](https://yadi.sk/i/O33gixd55tRMZw)
[tensorboard screenshot for mode='bilinear'](https://yadi.sk/i/HjuOVM7CPO1Dyw)

**Comment by edgarriba** [comment]
@darkAlert could you provide a short snippet showing this behaviour?

**Comment by darkAlert** [comment]
@edgarriba, sure

```
class MyModel(nn.Module):
	def __init__(self, out_channels, h, w, nearest=False):
		# Custom resnet:
        self.resnet = my_resnet(out_channels)

        # Image that will be warped:
        self.template = template       

        if nearest:
        	self.warper = kornia.HomographyWarper(h, w, mode='nearest')
        else:
            self.warper = kornia.HomographyWarper(h, w)

        # Regressor that outputs the 3x3 homography matrix:
        self.reg = nn.Sequential(
            nn.Linear(out_channels, 32),
            nn.ReLU(True),
            nn.Linear(32, 3 * 3)
        )
        self.reg[-1].weight.data.zero_()
        self.reg[-1].bias.data.copy_(torch.tensor([1, 0, 0, 0, 1, 0, 0, 0, 1], dtype=torch.float))

    def forward(self, x):
        x = self.resnet(x)
        theta = self.reg(x)
        warped = self.warp(theta, self.template)

        return warped
```

**Comment by darkAlert** [comment]
@edgarriba, sure

```
class MyModel(nn.Module):
	def __init__(self, out_channels, h, w, nearest=False):
		# Custom resnet:
		 self.resnet = my_resnet(out_channels)

		# Image that will be warped:
		self.template = template       

		 if nearest:
        		self.warper = kornia.HomographyWarper(h, w, mode='nearest')
		else:
            		self.warper = kornia.HomographyWarper(h, w)

                # Regressor that outputs the 3x3 homography matrix:
                self.reg = nn.Linear(out_channels, 3*3)
                self.reg[-1].weight.data.zero_()
                self.reg[-1].bias.data.copy_(torch.tensor([1, 0, 0, 0, 1, 0, 0, 0, 1], dtype=torch.float))

	def warp(self, theta, template):
		bs = theta.shape[0]
		template = template[0:bs]
		warped = self.warper(template, theta)

		return warped.squeeze(1)

	def forward(self, x):
		x = self.resnet(x)
                x = self.reg(x)
                theta = x.view(-1, 1, 3, 3)
                warped = self.warp(theta, self.template)
        
                return warped
```

Tensors have the following dimensions:
template = [bs, 1, 360, 640]
theta = [bs, 1, 3, 3]
warped = [bs, 1, 360, 640]

**Comment by shijianjian** [comment]
I think it happens to all the functions that used ```F.grid_sample```. You will get grad=0 if set mode to nearest.

https://discuss.pytorch.org/t/autograd-issue-with-f-grid-sample/76894

https://discuss.pytorch.org/t/does-the-nn-funtional-grid-sample-backwards-grad-to-grid/31736/3
